src/conn/conn_log.c

```diff
@@ -660,22 +662,30 @@ __log_wrlsn_server(void *arg)
 	WT_CONNECTION_IMPL *conn;
 	WT_DECL_RET;
 	WT_SESSION_IMPL *session;
+	int yield;
 
 	session = arg;
 	conn = S2C(session);
+	yield = 0;
 	while (F_ISSET(conn, WT_CONN_LOG_SERVER_RUN)) {
 		/*
 		 * Write out any log record buffers.
 		 */
-		WT_ERR(__wt_log_wrlsn(session));
-		WT_ERR(__wt_cond_wait(session, conn->log_wrlsn_cond, 10000));
+		WT_ERR(__wt_log_wrlsn(session, &yield));
+		/*
+		 * If __wt_log_wrlsn did work we want to yield instead of sleep.
+		 */
+		if (yield++ < 1000)
```

Comment:
First it restores the original code that was in the earlier logging code.  There are a lot of places that yield for 1000 loops and then sleep.  In this case, we've had complaints about worker threads consuming too much CPU.  I might be inclined to also restore the original 100000 `wt_cond_wait` timeout time as well.